encoding/test_encoding.py

```diff
@@ -170,6 +169,60 @@ def run_test(self, method, body_expected):
             self.assertEqual(len(server.requests), 2)
 
     def test(self):
+        self.start_all()
+        self.run_test("GET", True)
+
+
+class TestH2BodyDechunkingWithMovingBody(TestH2BodyDechunking):
+    """
+    Tempesta FW moves body during making headers if count
+    of skb fragments is exceeded. Later Tempesta FW  removes
+    flaged data from the body, so all pointers should be correct.
+    """
+
+    backends_template = [
+        {
+            "id": "backend",
+            "type": "deproxy",
+            "port": "8000",
+            "response": "static",
+            "response_content": "HTTP/1.1 200 OK\r\n"
+            "Content-type: text/html\r\n"
+            f"Last-Modified: {DATE}\r\n"
+            f"Date: {DATE}\r\n"
+            "a: " + "".join(random.choice(string.ascii_lowercase) for i in range(1000)) + "\r\n"
+            "b: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "c: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "d: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "e: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "r: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "q: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "y: " + "".join(random.choice(string.ascii_lowercase) for i in range(5000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(3000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(8000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(8000)) + "\r\n"
+            "z: " + "".join(random.choice(string.ascii_lowercase) for i in range(8000)) + "\r\n"
```

Comment:
Looks like like a new method to generation headers? 
```python3
def generate_header(header_name: str, header_value_n: int) -> str:
    return f"header_name: " + "".join(random.choice(string.ascii_lowercase) for i in range(header_value_n)) + "\r\n"
```

You can add it like class method before creation `backends_template` or like a global method in this file.